### ISSUE TITLE
Split up backends (TF, SameDiff, DL4J, etc) + 'platform' artifacts into different profiles?

### ISSUE THREAD

**Issue by AlexDBlack** [KonduitAI/konduit-serving#236]
Motivation: Suppose I'm serving a TF model on Linux CPU. Why do I need other operating systems, DL4J, ONNX, etc dependencies?

On the one hand, splitting this would make it more complex for users and releases.

The main benefit is the reduced binary size. Once we add TF, SameDiff, DL4J (with cuDNN), and ONNX all into the one JAR, we could be looking at a couple of GB total size.
This is further compounded by the fact that we use platform binaries - i.e., all OSs for things like ND4J, MLKDNN and TensorFlow.

**Comment by saudet** [comment]
It's already like that:
https://github.com/bytedeco/javacpp-presets/wiki/Reducing-the-Number-of-Dependencies
What's missing?

**Comment by agibsonccc** [comment]
@AlexDBlack  generally the javacpp binary size once we shrink it down to certain OSes isn't too bad. We would basically need to make it so you load an inference executioner as a loaded service or something like nd4j. If we want to do that, I'm not against it.

**Comment by AlexDBlack** [comment]
@Saudet we do have that which solves half the problem here, at least for our own builds
my concern is more from a usability perspective (needing your average Python user to know about javacpp platforms config isn't good)
maybe a simple script on top of the existing javacpp config would be fine for handling that concern though.

the "dl4j + onnx + tf + samediff all present in the jar but only using 1 of them" problem is my bigger concern though.

I haven't quite decided what we should do here, my main thing right now was simply to flag it for review/fixing in the future.

**Comment by saudet** [comment]
I would consider doing it in a similar fashion to how other platforms have, that is provide installers for single platforms, but also have repositories from which additional packages can be downloaded from when required.

**Comment by agibsonccc** [comment]
One thing we could do is a "konduit configure" maybe? Auto detect the OS and go through a series of steps of what people want? Use that to build the jar?

**Comment by AlexDBlack** [comment]
Right, "wizard-style" CLI to select things could work. i.e., walk user through selecting which OSes, hardware, etc to support. Then yes, simply convert that into which maven profiles to use.
